### PR TITLE
[Debugger] Add variable evaluation feature 

### PR DESCRIPTION
**Pull request by pranavan15**
## Purpose

Adding variable evaluation feature to the Ballerina debugger (Resolves https://github.com/ballerina-platform/ballerina-lang/issues/10090). This allows users to evaluate any variable while debugging the program.

### CODE REVIEW

File: bvm/ballerina-core/src/main/java/org/ballerinalang/util/debugger/ExpressionEvaluator.java [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+     * @param currentExecLine Current execution line.
+     * @param variableName    Name of the variable to be evaluated.
+     * @return Evaluated results.
+     */
+    public String evaluateVariable(WorkerExecutionContext ctx, LineNumberInfo currentExecLine, String variableName) {
+        int currentExecLineNum = currentExecLine.getLineNumber();
+
+        List<PackageVarInfo> globalVars = getPackageVariables(ctx);
+        for (PackageVarInfo var : globalVars) {
+            if (var.getName().equals(variableName)) {
```
[the remark this answers is not in the document]
Comment (pranavan15) [reply bvm/ballerina-core/src/main/java/org/ballerinalang/util/debugger/ExpressionEvaluator.java]: Global variables belong to the package and hence they are visible everywhere within the package from the very beginning of the program execution right? In the above scenario you mentioned, variable `a` would have its default value 0 when a debug point hits within the function `test()`. Once, `test` function gets executed a new value will be assigned to `a`. As global variables are always visible and have a value, scope becomes inappropriate here right?

[the remark this answers is not in the document]
Comment (lankavitharana) [reply bvm/ballerina-core/src/main/java/org/ballerinalang/util/debugger/ExpressionEvaluator.java]: so you mean if we hit a debug point in the first line (which is int a = test();)  even then we should see the string b variable?

[the remark this answers is not in the document]
Comment (lankavitharana) [reply bvm/ballerina-core/src/main/java/org/ballerinalang/util/debugger/ExpressionEvaluator.java]: in java this is the case, we can see variable "b" even when debub point is in "a" (b is null at that point) so let's go ahead with this for time being, we may need to think about better value than "null" (may be "uninitialized"?)

**Comment by hasithaa** [comment]
@pranavan15 have we done a code review session for this? If not could you schedule a review?

**Comment by pranavan15** [comment]
Closing until the addressing the code review changes

**Pull request closed without merging**